src/latexify/frontend.py

```diff
@@ -80,74 +95,67 @@ def get_latex(
         ).visit(tree)
 
 
-class LatexifiedFunction:
-    """Function with latex representation."""
+@overload
+def algorithmic(alg: Callable[..., Any], **kwargs: Any) -> output.LatexifiedAlgorithm:
+    ...
 
-    _fn: Callable[..., Any]
-    _latex: str | None
-    _error: str | None
 
-    def __init__(self, fn, **kwargs):
-        self._fn = fn
-        try:
-            self._latex = get_latex(fn, **kwargs)
-            self._error = None
-        except exceptions.LatexifyError as e:
-            self._latex = None
-            self._error = f"{type(e).__name__}: {str(e)}"
+@overload
+def algorithmic(
+    **kwargs: Any,
+) -> Callable[[Callable[..., Any]], output.LatexifiedAlgorithm]:
+    ...
 
-    @property
-    def __doc__(self):
-        return self._fn.__doc__
 
-    @__doc__.setter
-    def __doc__(self, val):
-        self._fn.__doc__ = val
+def algorithmic(
+    alg: Callable[..., Any] | None = None, **kwargs: Any
+) -> output.LatexifiedAlgorithm | Callable[
+    [Callable[..., Any]], output.LatexifiedAlgorithm
+]:
+    """Attach LaTeX pretty-printing to the given algorithm.
```

Comment:
It is unclear what kind of stuff the term "algorithm" points to. use "function".

```suggestion
    """Attach LaTeX pretty-printing to the given function.
```